dask/dataframe/methods.py

```diff
@@ -106,3 +109,62 @@ def pivot_count(df, index, columns, values):
     # make dtype deterministic, always coerce to np.float64
     return pd.pivot_table(df, index=index, columns=columns,
                           values=values, aggfunc='count').astype(np.float64)
+
+
+# ---------------------------------
+# concat
+# ---------------------------------
+
+def concat(dfs, axis=0, join='outer'):
+    """ Concatenate caring empty Series """
+
+    # can be removed after pandas 0.18.1 or later
+    # see https://github.com/pandas-dev/pandas/pull/12846
+    if PANDAS_VERSION >= '0.18.1':
+        return pd.concat(dfs, axis=axis, join=join)
+
+    # Concat with empty Series with axis=1 will not affect to the
+    # result. Special handling is needed in each partition
+    if axis == 1:
+        # becahse dfs is a generator, once convert to list
+        dfs = list(dfs)
+
+        if join == 'outer':
+            # outer concat should keep all empty Series
+
+            # input must include one non-empty data at least
+            # because of the alignment
+            first = [df for df in dfs if len(df) > 0][0]
+
+            def _pad(base, fillby):
+                if isinstance(base, pd.Series) and len(base) == 0:
+                    # use aligned index to keep index for outer concat
+                    return pd.Series([np.nan] * len(fillby),
+                                     index=fillby.index, name=base.name)
+                else:
+                    return base
+
+            dfs = [_pad(df, first) for df in dfs]
+        else:
+            # inner concat should result in empty if any input is empty
+            if any(len(df) == 0 for df in dfs):
+                dfs = [pd.DataFrame(columns=df.columns)
+                       if isinstance(df, pd.DataFrame) else
+                       pd.Series(name=df.name) for df in dfs]
+
+    return pd.concat(dfs, axis=axis, join=join)
+
+
+def merge(left, right, how, left_on, right_on,
+          left_index, right_index, indicator, suffixes,
+          default_left, default_right):
+
+    if not len(left):
+        left = default_left
+
+    if not len(right):
+        right = default_right
+
+    return pd.merge(left, right, how=how, left_on=left_on, right_on=right_on,
+                    left_index=left_index, right_index=right_index,
+                    suffixes=suffixes, indicator=indicator)
```

Comment:
I'm curious generally when we should move methods like these to methods.py and when we should keep them close to their dask.dataframe equivalents.  To me it seems convenient to keep them close to their main dask.dataframe functions because we are likely to change both together. 

Methods.py does exist for this purpose though.  I don't know what the policy should be.  @sinhrks what do you think?  @jcrist  any thoughts?